### DISCUSSION TITLE
is it possible to use floating-ui/react without `refs.setReference`

### DISCUSSION THREAD

**Discussion by wulftn** [floating-ui/floating-ui#2453] [Q&A]
I have a use case like this:

```
Library component:
const Button = (props) => {
  return <button {...props}></button>
}

My component:
<Button {...getReferenceProps()} ref={refs.setReference}></Button>
```

Commonly, some wrapper components do not expose the inner ref so it breaks the tooltip/popover position.
I can wrap the Button like 
```
<div  {...getReferenceProps()} ref={refs.setReference}>
 <Button></Button>
</div>
```
but the wrapper div is not good:
1. break CSS style
2. break floating-ui/react's default hooks like the `isSpaceIgnored` check in the `useClick`?
3. the reference element is disabled but the tooltip still shows.

is there a better workaround for my case?

**Comment by atomiks** [comment]
There isn't a way to get around it aside from the wrapper. The ref is used for measuring the reference element, adding listeners outside of React's event handler props, among other things.

> Commonly, some wrapper components do not expose the inner ref so it breaks the tooltip/popover position.

Why don't they? You should fix that because forwarding refs in reusable components is necessary in most cases.

**Reply by wulftn** [reply]
thank you for the suggestion.
I can see in some cases, the reusable components' ref is used for custom methods.
I think we will need an extra prop like `innerRef` to forward it in this case.
